### ISSUE TITLE
Lottie React build failure (eval is deprecated)

### ISSUE THREAD

**Issue by SpaceHolderdJs** [Gamote/lottie-react#98]
Hello Everyone!

I am using lottie-react with `"react": "^18.2.0",` and [Vite](https://vitejs.dev/) and facing the issue after running the `build` command.

Here is the output of the terminal

<img width="698" alt="image" src="https://github.com/Gamote/lottie-react/assets/62031924/97b622f9-f003-413a-931c-6c8fedd77df4">

I am wondering if there is any way to fix it, would be grateful for any support, thanks!

**Comment by pramesh07** [comment]
yeah am also facing the same issue due to use of eval am also using react 18.2.0 and vite

**Comment by patrickleogit** [comment]
I am facing the same issue with React 18.2.0, Typescript 4.9.5  and Vite 4.5.0

**Comment by Baalamurgan** [comment]
Same here with Next 13.4.19, Typescript 5.2.2, React 18.2.0

**Comment by Supcar27** [comment]
I am also facing this issue with React 18.2

**Comment by SteveBehm** [comment]
The company I work for has a Next.js/React app and I have experienced this issue during build and I have been unable to resolve it.

If anyone has a workaround for the time being that would be helpful.

<img width="1025" alt="Screenshot 2023-11-09 at 1 22 06 PM" src="https://github.com/Gamote/lottie-react/assets/96929704/cdd75b69-aeeb-4088-9033-a5758d393526">

**Comment by w-bouzazi** [comment]
Same issue on my end.

**Comment by kevinvugts** [comment]
Is this something that can be picked up by you @Gamote  ?

**Comment by Gamote** [comment]
That message is a warning from Vite's bundler, not a failure: it reports that a dependency contains `eval`, and the build completes. The `eval` is lottie-web's expression evaluator, the part that runs After Effects expressions.

v3 gives you a way to not ship it. If your animations use no expressions, import `LottieLight` (or `useLottieLight`) instead of `Lottie`. It runs the `lottie_light` build of the engine, which contains no `eval` at all, so the warning disappears, and it also matters under a strict Content Security Policy. Importing it never pulls the full engine into your bundle; the build measures that.

```tsx
import { LottieLight } from "lottie-react";

export function Hero() {
  return <LottieLight src="/hero.json" autoplay loop />;
}
```

If your animations do use expressions, keep `Lottie` (or `LottieSvg`) and the warning stays; it is advisory. The three builds and what each drops: https://lottiereact.com/docs/animation/renderers-and-smaller-builds

The other error in this thread, `document is not defined` while Next prerenders (@Baalamurgan, @SteveBehm), was a separate lottie-web fault fixed in 5.13.0, which v3 and 2.4.2 both require; details in #123.

Closing.

> [!NOTE]
> ### 🎉 lottie-react v3 is out
>
> A rewrite, and the things people asked for the most are in: `src` fetches a URL or takes the object, `loop`, `speed` and `direction` are reactive, `seek` moves by frame, marker, percent or seconds, an accessible control bar, scroll and in-view interactions, `LottieSvg` and `LottieLight` builds, server rendering with no dynamic import, React 18 and 19, only what you import ships, and more.
>
> [Docs and live examples](https://lottiereact.com) · [Migration guide](https://lottiereact.com/docs/migration), every v2 surface mapped to its v3 home.
